test/plots/scatterPlotTests.ts

```diff
@@ -2,395 +2,386 @@
+
+        svg.remove();
+      });
+
+      it("can retrieve Entities in a certain range", () => {
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesIn({ min: xScale.scale(1), max: xScale.scale(1) },
+                                      { min: yScale.scale(1), max: yScale.scale(1) });
+
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
+        svg.remove();
+      });
+
+      it("does not return Entities whose center lies outside the range", () => {
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesIn({ min: xScale.scale(1.001), max: xScale.scale(1.001) },
+                                      { min: yScale.scale(1.001), max: yScale.scale(1.001) });
+
+        assert.lengthOf(entities, 0, "no Entities retrieved");
+
+        svg.remove();
+      });
+
+      it("can retrieve Entities in a certain bounds", () => {
+        plot.renderTo(svg);
+        let entities = plot.entitiesIn({ topLeft: {
+                                          x: xScale.scale(1),
+                                          y: yScale.scale(1)
+                                        },
+                                        bottomRight: {
+                                          x: xScale.scale(1),
+                                          y: yScale.scale(1)
+                                        }});
+
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
+        svg.remove();
+      });
+
+      it("can retrieve Entities centered at a given Point", () => {
+        plot.removeDataset(dataset2);
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesAt({ x: xScale.scale(1), y: yScale.scale(1) });
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
+        svg.remove();
+      });
+
+      it("determines whether an Entity contains a given Point by its position and size", () => {
+        plot.size(10);
+        plot.removeDataset(dataset2);
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesAt({ x: xScale.scale(1) + 5, y: yScale.scale(1) - 5});
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
+        plot.size(6);
+        entities = plot.entitiesAt({ x: xScale.scale(1) + 5, y: yScale.scale(1) - 5});
+        assert.lengthOf(entities, 0, "none of the Entities is retrieved");
+        svg.remove();
+      });
+
+      it("returns all Entities containing a given Point across all Datasets", () => {
+        dataset.data([{ x: 0, y: 0 }, { x: 200, y: 200 }]);
```

Comment:
Don't change the data of the dataset, instead add a new dataset with the new data. This stacks with the "remove dataset2 from beforeEach"